pkg/distro/rhel9/images.go

```diff
@@ -698,3 +708,25 @@ func initialSetupKickstart() *fsnode.File {
 	}
 	return file
 }
+
+// fixes: https://github.com/osbuild/images/issues/352
+// Creates a unit file that create mountpoints if it doesnot exits.
+// This ensures that the filesystem created using image builder continues to work after upgrade.
+func createFilesystemMounpointService(customizations *blueprint.Customizations) *fsnode.File {
```

Comment:
In general, using custom files for things which are not one-off or a special case is highly discouraged in favor of explicit osbuild stage.

There is already https://github.com/osbuild/osbuild/blob/main/stages/org.osbuild.systemd.unit.

While it was developed mainly to create service unit dropins, The name of the stage is a bit unfortunate, since ideally it should be `org.osbuild.systemd.unit.dropin` or something similar. That said, I think that the `dropin` stage option could be made optional with the behavior of the stage changing to create the actual unit file, instead of the unit dropin. The stage will require additional options to be able to cover your use case, but I think that these will prove useful in the future for other scenarios as well.

Lastly, the function should then end up in the `osbuild` package and not in a distro definition.

Reply:
@thozza I think renaming the stage  `org.osbuild.systemd.unit` ->  `org.osbuild.systemd.unit.dropin` will make sense as it will correctly depict of whats the stage doing,  and modifying 'org.osbuild.systemd.unit' to create systemd units.

Reply:
We can't rename existing stages and replace them when incompatible implementation. It is not an option, sorry.

Reply:
done.